src/components/base-control.js

```diff
@@ -39,24 +40,19 @@ const defaultProps = {
   captureDoubleClick: true
 };
 
-const contextTypes = {
-  viewport: PropTypes.instanceOf(WebMercatorViewport),
-  isDragging: PropTypes.bool,
-  eventManager: PropTypes.object
-};
-
 /*
  * PureComponent doesn't update when context changes.
  * The only way is to implement our own shouldComponentUpdate here. Considering
  * the parent component (StaticMap or InteractiveMap) is pure, and map re-render
  * is almost always triggered by a viewport change, we almost definitely need to
  * recalculate the marker's position when the parent re-renders.
  */
-export default class BaseControl extends Component {
+export default class BaseControl extends PureComponent {
 
   constructor(props) {
     super(props);
 
+    this.context = {};
```

Comment:
This may be broken because `context` field is used by old context api. I would use another name.